target/linux/realtek/files-6.12/drivers/net/pcs/pcs-rtl-otto.c

```diff
@@ -863,20 +863,22 @@ static int rtpcs_930x_sds_config_pll(struct rtpcs_serdes *sds,
 	else
 		return -ENOTSUPP;
 
-	if (!neighbor_mode)
-		pll = speed == RTSDS_930X_PLL_10000 ? RTSDS_930X_PLL_LC : RTSDS_930X_PLL_RING;
-	else if (speed == neighbor_speed) {
-		speed_changed = false;
-		pll = neighbor_pll;
-	} else if (neighbor_pll == RTSDS_930X_PLL_RING)
+	/* 10GBASE-R always requires LC PLL */
+	if (speed == RTSDS_930X_PLL_10000) {
 		pll = RTSDS_930X_PLL_LC;
-	else if (speed == RTSDS_930X_PLL_10000) {
-		pr_info("%s: SDS %d needs LC PLL, reconfigure SDS %d to use ring PLL\n",
-			__func__, sds->id, nb_sds->id);
-		rtpcs_930x_sds_reconfigure_pll(nb_sds, RTSDS_930X_PLL_RING);
+		if (neighbor_mode && neighbor_pll == RTSDS_930X_PLL_LC &&
+		    neighbor_speed != RTSDS_930X_PLL_10000) {
+			pr_info("%s: SDS %d needs LC PLL, reconfigure SDS %d to use ring PLL\n",
+				__func__, sds->id, nb_sds->id);
+			rtpcs_930x_sds_reconfigure_pll(nb_sds, RTSDS_930X_PLL_RING);
+		}
+	} else if (!neighbor_mode) {
+		pll = RTSDS_930X_PLL_RING;
+	} else if (neighbor_pll == RTSDS_930X_PLL_RING) {
 		pll = RTSDS_930X_PLL_LC;
-	} else
+	} else {
```

Comment:
the preferred Linux kernel coding style is keeping this without brackets as it's a single line

> Do not unnecessarily use braces where a single statement will do.

https://www.kernel.org/doc/html/v4.10/process/coding-style.html

Reply:
Also can't we join `else if (!neighbor_mode)` and `else` into one statement `else`?

`neighbor_pll == RTSDS_930X_PLL_RING` implies that neighbor_mode is true, doesn't it?